examples/Controllers/MPC/mpcFcn.m

```diff
@@ -0,0 +1,214 @@
+  %   C-Mex counterpart: mdlOutputs
+  %
+  block.RegBlockMethod('Outputs', @Outputs);
+ 
+%endfunction
+
+function Outputs(block)
+
+global mpcStruct;
+
+% This entire block is the setup and execution of mpcStruct
+
+Sx = mpcStruct.MPCSetup.Sx;    % Coeff matrix for current state, x(k)
+Su = mpcStruct.MPCSetup.Su;    % Coeff matrix for dFpto, u(k)
+Sv = mpcStruct.MPCSetup.Sv;    % Coeff matrix for Fe, v(k)
+Q = mpcStruct.MPCSetup.Q;
+HpInK = mpcStruct.MPCSetup.HpInk;
+x_k = block.InputPort(1).Data;
+
+v_arrow_k = block.InputPort(2).Data;
+
+%% Create cost matrices for mpcStructStruct objective function
+H = mpcStruct.MPCSetup.H;
+f = Su'*Q*(Sx*x_k+Sv*v_arrow_k);
+%mpcStruct.f=f;
+
+%% Hessian check for convex solver 
+if issymmetric(H) == 0
+    H_Memory = H;
+    checkH=(H+H')/2;
+    if abs(H-H_Memory) < 1e-15
+        %display('H was symetric to within a 1e-15 tolerance');
+        H=(H+H')/2;
+    else 
+        hello = 1;
+    end
+end
+
+if all(eig(H) >= 0) == 0
+    eig(checkH)
+    issymmetric(checkH)
+    display('PSD Condition violated on checkH!')
+end
+
+
+
+%% Create constraint matrices for mpcStruct
+
+I = eye(mpcStruct.MPCSetup.HpInk+1);%length of prediction horizon
+Zero = zeros(mpcStruct.MPCSetup.HpInk+1);
+
+% Create y_max vector [dZb,max; Zb,max; Fpto,max] to Hp
+iterationCounter = 1;
+for i = 1:1:((HpInK)*mpcStruct.modelPredictiveControl.yLen)
+
+   % Single pod
+         if iterationCounter == 1
+            y_arrow_upperbound(i,1) = mpcStruct.modelPredictiveControl.maxVel;
+         elseif iterationCounter == 2
+            y_arrow_upperbound(i,1) = mpcStruct.modelPredictiveControl.maxPos;
+         elseif iterationCounter == 3
+            y_arrow_upperbound(i,1) = mpcStruct.modelPredictiveControl.maxPTOForce;
+         end
+    
+    iterationCounter = iterationCounter + 1;
+    if mod(i,mpcStruct.modelPredictiveControl.yLen) == 0
+        iterationCounter = 1;
+    end
+end
+
+% Simple dFpto max
+u_arrow_upperbound = mpcStruct.modelPredictiveControl.maxPTOForceChange*ones(HpInK+1,1); % dFpto upper limit
+
+% REACTIVE POWER LIMIT: Determine the sign to limit the velocity at k+1
+% based on what it is here at time k
+if x_k(1) >= 0
+    signOfReactivePowerLimit = -1; % If v is pos, Fpto must be negative
+else 
+    signOfReactivePowerLimit = 1;
+end
+%mpcStruct.y_arrow_upperbound = y_arrow_upperbound;
+
+A = [...
+    I; ...
+    -I; ...
+    Su; ...
+    -Su; ...
+    ];
+
+B = [...
+    u_arrow_upperbound; ...
+    u_arrow_upperbound; ...
+    y_arrow_upperbound-Sx*x_k-Sv*v_arrow_k; ...
+    y_arrow_upperbound+Sx*x_k+Sv*v_arrow_k];
+
+
+%% Run quadprog
+
+if mpcStruct.MPCSetup.currentIteration > mpcStruct.modelPredictiveControl.Ho + mpcStruct.modelPredictiveControl.order % Only run mpcStruct after buffer is full, otherwise Fe predictions are not accurate
+    options = optimoptions('quadprog','Algorithm','interior-point-convex','Display','off', 'MaxIter', 600); % Can't set x0 w/ interior-point-convex 
```

Comment:
Quadprog is part of the optimization toolbox and will not run with base MATLAB.

Reply:
With this being moved to the applications case, the extra toolbox shouldn't be an issue from what I understand.